Sources/SimplenoteSearch/Query/SearchQuery.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+// MARK: SearchQuery: wraps a search text
+//
+@objc
+public final class SearchQuery: NSObject {
+
+    /// Original search text
+    ///
+    @objc
+    public let searchText: String
+
+    /// A boolean value indicating if the query is empty
+    ///
+    @objc
+    public var isEmpty: Bool {
+        return items.isEmpty
+    }
+
+    /// Ordered list of query items found in a search text
+    ///
+    private(set) public var items: [SearchQueryItem] = []
+
+    /// Tags found in a search text
+    ///
+    @objc
+    public var tags: [String] {
+        return items.compactMap {
+            guard case let .tag(value) = $0, !value.isEmpty else {
+                return nil
+            }
+
+            return value
+        }
+    }
+
+    /// Keywords found in a search text
+    ///
+    @objc
+    public var keywords: [String] {
+        return items.compactMap {
+            guard case let .keyword(value) = $0 else {
+                return nil
+            }
+
+            return value
+        }
+    }
+
+    /// Init with a search text
+    ///
+    @objc
+    public init(searchText: String) {
+        self.searchText = searchText
+        super.init()
+        parse()
+    }
+
+    private func parse() {
+        let keywords = searchText.trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: .whitespaces)
+
+        for keyword in keywords where keyword.isEmpty == false {
+            guard let tag = keyword.lowercased().suffix(afterPrefix: String.searchOperatorForTags.lowercased()) else {
+                items.append(.keyword(keyword))
+                continue
+            }
+
+            items.append(.tag(tag))
```

Comment:
By skipping `.append` whenever `tag.isEmpty` in this spot, you can drop 2-3 checks in NSPredicate